python/cudf/cudf/core/dtypes.py

```diff
@@ -165,6 +165,9 @@ def __repr__(self):
         else:
             return f"ListDtype({self.element_type})"
 
+    def __hash__(self):
+        return id(self)
```

Comment:
I think this could be problematic, as it will be unique to every object, even when they are equal. We should compute a hash value that is equal for objects that compare equal (https://docs.python.org/3/reference/datamodel.html#object.__hash__).

Reply:
Updated the hash implementation to reply on pyarrow hash value to be consistent.